OpenRA.Mods.TS/UtilityCommands/LegacyTilesetImporter.cs

```diff
@@ -77,9 +97,14 @@ public void Run(ModData modData, string[] args)
 
 						using (var s = modData.DefaultFileSystem.Open(templateFilename))
 						{
-							Console.WriteLine("\tTemplate@{0}:", templateIndex);
-							Console.WriteLine("\t\tCategory: {0}", sectionCategory);
-							Console.WriteLine("\t\tId: {0}", templateIndex);
+							data.AppendLine("\tTemplate@{0}:".F(templateIndex));
+							data.AppendLine("\t\tCategory: {0}".F(sectionCategory));
+							if (categoriesUsage.ContainsKey(sectionCategory))
```

Comment:
And replace this with an unconditional addition to the set (sets can't contain duplicates).